ui/src/warp_runner/manager/commands/constellation_commands.rs

```diff
@@ -64,3 +88,184 @@ fn initialize_items(warp_storage: &mut warp_storage) -> Result<uplink_storage, E
+                                    )
+                                }
+                            }
+                        }
+                        Progression::ProgressComplete { name, total } => {
+                            log::info!(
+                                "{name} has been uploaded with {} MB",
+                                total.unwrap_or_default() / 1024 / 1024
+                            );
+                        }
+                        Progression::ProgressFailed {
+                            name,
+                            last_size,
+                            error,
+                        } => {
+                            log::info!(
+                                "{name} failed to upload at {} MB due to: {}",
+                                last_size.unwrap_or_default(),
+                                error.unwrap_or_default()
+                            );
+                        }
+                    }
+                }
+                match set_thumbnail_if_file_is_image(warp_storage.clone(), filename.clone()).await {
+                    Ok(success) => log::info!("{:?}", success),
+                    Err(error) => log::error!("Error on update thumbnail: {:?}", error),
+                }
+                log::info!("{:?} file uploaded!", &filename);
+            }
+            Err(error) => log::error!("Error when upload file: {:?}", error),
+        }
+    }
+    get_items_from_current_directory(warp_storage)
+}
+
+fn verify_duplicate_name(
+    current_directory: Directory,
+    filename: String,
+    file_pathbuf: PathBuf,
+) -> String {
+    let mut count_index_for_duplicate_filename = 1;
+    let mut new_file_name = filename.clone();
+    let original = filename;
+    loop {
+        if !current_directory.has_item(&new_file_name) {
+            break;
+        }
+        let file_extension = file_pathbuf
+            .extension()
+            .and_then(OsStr::to_str)
+            .map(str::to_string);
+        let file_stem = file_pathbuf
+            .file_stem()
+            .and_then(OsStr::to_str)
+            .map(str::to_string);
+
+        new_file_name = match (file_stem, file_extension) {
+            (Some(file_stem), Some(file_extension)) => {
+                format!("{file_stem} ({count_index_for_duplicate_filename}).{file_extension}")
+            }
+            _ => format!("{original} ({count_index_for_duplicate_filename})"),
+        };
+
+        log::info!("Duplicate name, changing file name to {}", new_file_name);
+        count_index_for_duplicate_filename += 1;
+    }
+    new_file_name
+}
+
+async fn set_thumbnail_if_file_is_image(
+    warp_storage: warp_storage,
+    filename_to_save: String,
+) -> Result<String, Box<dyn std::error::Error>> {
+    let item = warp_storage
+        .current_directory()?
+        .get_item(&filename_to_save)?;
+    let parts_of_filename: Vec<&str> = filename_to_save.split('.').collect();
```

Comment:
if a file name has no extension, i wonder if this will yield the file name instead. perhaps a `Path::extension()` should be used instead of  `name.split('.')`